cs/src/Connections/RelayTunnelConnector.cs

```diff
@@ -174,10 +174,14 @@ public RelayTunnelConnector(IRelayClient relayClient)
                 {
                     throw exception;
                 }
-                else if (retryingArgs.Delay >= TimeSpan.Zero)
+                else if ((int)retryingArgs.Delay.TotalMilliseconds > 0)
                 {
                     attemptDelayMs = (int)retryingArgs.Delay.TotalMilliseconds;
```

Comment:
Perhaps it should be `Math.Max((int)retryingArgs.Delay.TotalMilliseconds, RetryInitialDelayMs)`. To prevent the case when the event handler returns very small delay, that either rounds to 0 milliseconds (int), or is in milliseconds range, and that causes the next delays to be very small too.

Reply:
If it is less than 1ms, then the `(int)` cast will convert it to 0, and it will be treated as no delay, and will not update `attemptDelayMs`.

I don't see a reason why it can't be less than `RetryInitialDelayMs`. If the app's intent is to spam retries, then it can simply cancel this retry and immediately retry at a higher layer. So I don't think there's any point in trying to prevent that.